### DISCUSSION TITLE
Performance issue/using as XYZ server

### DISCUSSION THREAD

**Discussion by alturic** [developmentseed/titiler#523] [Q&A]
Hi all. I'm new to Python and I suppose my first question is, how exactly should you run the TiTiler server? I use it as an XYZ server for Google Maps, and I used the basic example on the docs, but it seems odd to me to run it via hypercorn using something like `nohup hypercorn ...` at the command line?

Secondly, I'm created web-optimized COG's via riogeo, and I store my TIFFs on the same physical server as the script, but I get an amazing amount of `not recognized as a supported file format` messages. Is there anything specific I can look into regarding this?

The other question I have, it seems the longer the script (again, ran with hypercorn) stays running the... slower it seems to become. Is there anything specific that could cause this?

**Comment by vincentsarago** [comment]
> Hi all. I'm new to Python and I suppose my first question is, how exactly should you run the TiTiler server? I use it as an XYZ server for Google Maps, and I used the basic example on the docs, but it seems odd to me to run it via hypercorn using something like nohup hypercorn ... at the command line?

I'm not sure to understand what's odd? 

> Secondly, I'm created web-optimized COG's via riogeo, and I store my TIFFs on the same physical server as the script, but I get an amazing amount of not recognized as a supported file format messages. Is there anything specific I can look into regarding this?

update `rasterio>=1.3` and those message will disappear

> The other question I have, it seems the longer the script (again, ran with hypercorn) stays running the... slower it seems to become. Is there anything specific that could cause this?

We've had report about memory-leak in the past
- https://github.com/tiangolo/fastapi/issues/4649
- https://github.com/developmentseed/titiler/issues/12
- https://github.com/tiangolo/fastapi/issues/596

But I'm not sure it will slow down the requests it self

**Comment by alturic** [comment]
I guess it seems strange to run a command like this on the CLI and then just close the terminal.

`nohup hypercorn --worker-class trio --workers 8 main:app --bind 127.0.0.1:8888 &`

I suppose you can make a daemon and such. Just was curious (again being new to Python and Python servers ultimately) if that was "normal".

In terms of the memory leak, I *thought* it could be something with switching over to swap which is what caused it to degrade after awhile.

Thanks for the replies!

**Comment by alturic** [comment]
@vincentsarago I'm using rasterio 1.3.0, on the tile server and I'm still getting:

`ERROR 4: `/vsimem/5b7a32a9-e10e-4ae2-aaa7-46532a8ad5ab/5b7a32a9-e10e-4ae2-aaa7-46532a8ad5ab.tif' not recognized as a supported file format.`

Is there a reason it's complaining about .tif's? All of the source files are .tiff.

**Reply by vincentsarago** [reply]
@alturic can you try 1.3.1 (https://github.com/rasterio/rasterio/blob/main/CHANGES.txt#L44)
